galsim/zernike.py

```diff
@@ -0,0 +1,398 @@
+                self.eps == other.eps and
+                self.diam == other.diam)
+
+    def __hash__(self):
+        return hash(("galsim.Zernike", tuple(self.coef.ravel()), self.eps, self.diam))
+
+    def __repr__(self):
+        out = "galsim.zernike.Zernike("
+        out += repr(self.coef)
+        if self.eps != 0.0:
+            out += ", eps={}".format(self.eps)
+        if self.diam != 2.0:
+            out += ", diam={}".format(self.diam)
+        out += ")"
+        return out
+
+
+def zernikeRotMatrix(jmax, theta):
+    """Construct Zernike basis rotation matrix.  This matrix can be used to convert a set of Zernike
+    polynomial series coefficients expressed in one coordinate system to an equivalent set of
+    coefficients expressed in a rotated coordinate system.  For example:
+
+        >>> Z = Zernike(coefs)
+        >>> R = zernikeRotMatrix(jmax, theta)
+        >>> rotCoefs = np.dot(R, coefs)
+        >>> Zrot = Zernike(rotCoefs)
+        >>> Z.evalPolar(r, th) == Zrot.evalPolar(r, th + theta)
+
+    Note that not all values of `jmax` are allowed.  For example, jmax=2 raises an Exception, since
+    a non-zero Z_2 coefficient will in general rotate into a combination of Z_2 and Z_3
+    coefficients, and therefore the needed rotation matrix requires jmax=3.  If you run into this
+    kind of Exception, raising jmax by 1 will eliminate it.
+
+    @param jmax  Maximum Zernike index (in the Noll convention) over which to construct matrix.
+    @param theta  Angle of rotation in radians.
+    @returns  Matrix of size [jmax, jmax].
+    """
+    # Use formula from Tatulli (2013) arXiv:1302.7106v1
+
+    # Note that coefficients mix if and only if they have the same radial index n and the same
+    # absolute azimuthal index m.  This means that to construct a rotation matrix, we need for both
+    # m's in a pair {(n, m), (n, -m)} to be represented, which places constraints on size.
+    # Specifically, if the final Zernike indicated by size includes only one part of the pair, then
+    # the rotation would mix coefficients into the element (size+1).  We simply disallow this here.
+
+    n_jmax, m_jmax = _noll_to_zern(jmax)
+    # If m_jmax is zero, then we don't need to check the next element to ensure we have a complete
+    # rotation matrix.
+    if m_jmax != 0:
+        n_jmaxp1, m_jmaxp1 = _noll_to_zern(jmax+1)
+        if n_jmax == n_jmaxp1 and abs(m_jmaxp1) == abs(m_jmax):
+            raise ValueError("Cannot construct Zernike rotation matrix of size {}".format(jmax))
+
+    R = np.zeros((jmax, jmax), dtype=np.float64)
+    for i in range(jmax):
+        ni, mi = _noll_to_zern(i+1)
+        for j in range(max(0, i-1), min(i+2, jmax)):
+            nj, mj = _noll_to_zern(j+1)
+            if ni != nj:
+                continue
+            if abs(mi) != abs(mj):
+                continue
+            if mi == mj:
+                R[i, j] = np.cos(mj * theta)
+            elif mi == -mj:
```

Comment:
This can just be else.  The `abs(mi) != abs(mj)` above means that this is the only option left.